server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -858,7 +880,8 @@ public void relocated(
             indexShardOperationPermits.blockOperations(30, TimeUnit.MINUTES, () -> {
                 forceRefreshes.close();
 
-                boolean syncTranslog = isRemoteTranslogEnabled() && Durability.ASYNC == indexSettings.getTranslogDurability();
+                boolean syncTranslog = (isRemoteTranslogEnabled() || this.isMigratingToRemote())
```

Comment:
i am guessing that we will upload the translog on migrating primary and hence this condition?

Reply:
Yes.